quad/src/qag_par.rs

```diff
@@ -266,22 +268,28 @@ impl QagPar {
         while last < self.limit {
             let mut to_process = vec![];
             let mut err_sum = 0.0;
+            let mut old_result = vec![0.0; n];
 
             while to_process.len() < 128 && heap.len() != 0 {
                 let old_interval = heap.pop().unwrap();
                 let ((x, y), old_err) = (old_interval.interval, old_interval.err);
+                if x.abs().max(y.abs())
+                    <= (1.0 + 100.0 * EPMACH) * (((x + y) / 2.0).abs() + 1000.0 * UFLOW)
+                {
+                    return QagIntegratorResult::new_error(ResultState::BadFunction);
+                }
                 let old_res = interval_cache
                     .remove(&(Myf64 { x }, Myf64 { x: y }))
                     .unwrap();
                 err_sum += old_err;
-                sub_vec(&mut result, &old_res);
+                add_vec(&mut old_result, &old_res);
                 to_process.push((x, y));
                 if err_sum > abserr - errbnd / 8.0 {
                     break;
                 }
             }
 
-            abserr -= err_sum;
+            //abserr -= err_sum;
```

Comment:
Why is this piece of code commented? If not needed, remove it, but if needed uncomment it :)